odoo/addons/base/views/ir_qweb_widget_templates.xml

```diff
@@ -28,12 +28,24 @@
                 <i t-if="not options.get('no_marker')" class="fa fa-map-marker fa-fw" role="img" aria-label="Address" title="Address"/>
                 <span class="d-block w-100 lh-sm" itemprop="streetAddress" t-esc="address"/>
             </div>
-            <div t-if="city and 'city' in fields" t-attf-class="d-flex align-items-baseline gap-1">
-                <i t-if="not options.get('no_marker')" class="fa fa-map-marker fa-fw" role="img" aria-label="Address" title="Address"/>
+            <div t-if="'city' in fields" t-attf-class="d-flex align-items-baseline gap-1">
+                <t t-set="has_city" t-value="city"/>
+                <t t-set="has_state" t-value="object.state_id and object.country_id.state_required"/>
+                <t t-set="has_country" t-value="country_id"/>
+                <t t-set="location" t-value="[]"/>
+                <i t-if="not options.get('no_marker') and (has_city or has_state or has_country)" class="fa fa-map-marker fa-fw" role="img" aria-label="Address" title="Address"/>
                 <span>
                     <div>
-                        <span itemprop="addressLocality" t-esc="city"/>,
-                        <span itemprop="addressCountry" t-esc="country_id"/>
```

Comment:
Well it could work but like I said above I was thinking something much simpler as we don't really need all the logic from event.
Perhaps something like this instead ?
```xml
<span itemprop="addressLocality" t-esc="city"/>,
<t t-if="object.state_id and object.country_id.state_required">
    <span itemprop="addressState" t-esc="object.state_id.code"/>,
</t>
<span itemprop="addressCountry" t-esc="country_id"/>
```

We don't really need to change much here, so just adding the state code with some condition should be enough imo.
And, as a bonus, it would only add 3 lines :slightly_smiling_face: